### PR TITLE
Revert Manage Themes dialog group/theme name input change

### LINKED ISSUE #2880: Revert design change on Manage Themes dialog
Sometimes in 2.0 we introduced a design change to the Manage themes dialog that made the whole thing a bit less user friendly. Let's change this back:

- Move the input to the left of the theme name input (this will fail for long theme names or small window sizes, but we can tackle this in a more elaborate UX fix when we align with Studio)
- Make the button a Button with variant secondary
- Add a dropdown prop (arrow down) when there's other groups to pick from. If no other groups exist, dont add the dropdown arrow.

Image

### PR DESCRIPTION
**Pull request by macintoshhelper**
### Why does this PR exist?

Fixes #2880  

### Testing this change

1. Go to the Manage Themes modal by clicking `Theme: None` or `Theme: {theme_name_here}`
2. Click `Manage Themes`
3. Click the `+ New Theme` button
4. Check visually that the new UI is being used
5. Try entering a group name and theme name

### Additional Notes (if any)

- Also fixes a bug around empty group name populating the input when renaming the group name
- Added text input placeholders for `Add group` and `Theme name`

| **Before** | **After** |
|--------|--------|
| image | image |
| image | image | 
| image | image | 
| image | image | 
| image | image |

### CODE REVIEW

File: packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/CreateOrEditThemeForm.tsx [hunk 1]
```diff
@@ -175,7 +178,7 @@ export const CreateOrEditThemeForm: React.FC<React.PropsWithChildren<React.Props
       )}
       <Stack direction="column" gap={1}>
         {activeTab === ThemeFormTabs.SETS && (
-        <Stack direction="column" gap={1} css={{ padding: '0 $4 $3' }}>
+        <Stack direction="column" gap={1} css={{ padding: '$3 $4 $3' }}>
```
Comment (macintoshhelper) [inline packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/CreateOrEditThemeForm.tsx:181]: Adds a little spacing so the group buttons aren't touching the `1px` border line.

image

File: packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/CreateOrEditThemeForm.tsx [hunk 2]
```diff
@@ -118,6 +117,7 @@ export const CreateOrEditThemeForm: React.FC<React.PropsWithChildren<React.Props
                 autofocus
                 data-testid="create-or-edit-theme-form--group--name"
                 {...register('group')}
+                placeholder="Add group"
```
Comment (cuserox) [inline packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/CreateOrEditThemeForm.tsx:120]: Maybe use `createNewGroup` as it's a translated string? Or anything similar.

File: packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/CreateOrEditThemeForm.tsx [hunk 3]
```diff
@@ -155,13 +155,16 @@ export const CreateOrEditThemeForm: React.FC<React.PropsWithChildren<React.Props
...
           }
-              <Box>/</Box>
+              <Box css={{ margin: '0 $3' }}>/</Box>
+            </Stack>
+            <Stack direction="row" gap={1} align="center" css={{ flexGrow: 1 }}>
+              <Input
+                full
+                data-testid="create-or-edit-theme-form--input--name"
+                {...register('name', { required: true })}
+                placeholder="Theme name"
```
Comment (cuserox) [inline packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/CreateOrEditThemeForm.tsx:165]: Similar comment about this placeholder being a translated string if possible

File: packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/ThemeGroupDropDownMenu.tsx [hunk 4]
```diff
@@ -41,14 +41,24 @@ export const ThemeGroupDropDownMenu: React.FC<React.PropsWithChildren<React.Prop
...
           ) : (
             <Box css={{ display: 'flex', alignItems: 'center', gap: '$2' }}>
-              <IconPlus />
-              Add&nbsp;group
+              <Button
+                data-testid="button-manage-themes-modal-group-dropdown"
+                variant="secondary"
+                icon={<IconPlus />}
+                size="small"
+                css={{ display: 'flex', alignItems: 'center', height: '28px' }}
```
Comment (cuserox) [inline packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/ThemeGroupDropDownMenu.tsx:57]:
```suggestion
                css={{ display: 'flex', alignItems: 'center', height: $controlMedium }}
```

Comment (cuserox) [reply packages/tokens-studio-for-figma/src/app/components/ManageThemesModal/ThemeGroupDropDownMenu.tsx:57]: Actually, none of this CSS is needed? Just tried disabling it and there's other styling that takes over, and is the same as this.

**Review by cuserox — APPROVED** [review]
A few minor comments. Looks good :)

**Comment by six7** [comment]
@macintoshhelper would you want to make these changes in a follow-up PR? I'll merge this in for rc8 given the requested changes dont change behavior too much 👍

**Pull request merged**